test/unit/foreman/logging_test.rb

```diff
@@ -33,7 +33,14 @@ def test_logger_supports_silence
     logger = Foreman::Logging.logger('test_logger')
     appender = ::Logging::Appenders.string_io('buffer')
     logger.appenders = [appender]
-    logger.silence { |log| log.warn("Should not be logged") }
+    logger.silence do |log|
+      log.debug("Must not be logged")
+      assert_equal 3, log.local_level
+      assert_equal 0, log.level
+    end
+    logger.silence_logger do |log|
+      log.warn("Must not be logged as well")
+    end
```

Comment:
I wonder if we could get something like `StringIO` as the backend of the logger, so that we could get it's content at the end and check, which strings got and got not logged?

Reply:
We already do that in the test.

Reply:
I see, perhaps we should however test that at least something gets logged if the level is high enough

Reply:
I added two positive tests and rebased.